### PR TITLE
update critical infra to use tf0.12

### PR DESCRIPTION
**Pull request by jamesgorrie**
As I move to putting the Calm VHS in here.

### CODE REVIEW

File: infrastructure/critical/modules/rds/security_groups.tf [hunk 1]
```diff
@@ -0,0 +1,30 @@
+resource "aws_security_group" "database_sg" {
+  description = "controls direct access to application instances"
```
Comment (alexwlchan) [inline infrastructure/critical/modules/rds/security_groups.tf]: Does it actually, or have you copy/pasted this without checking it?

File: infrastructure/critical/modules/rds/security_groups.tf [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  vpc_id      = var.vpc_id
+  name        = "${var.database_name}_sg"
```
Comment (alexwlchan) [inline infrastructure/critical/modules/rds/security_groups.tf]: Maybe:

```suggestion
  name        = "${var.database_name}_database_sg"
```

File: infrastructure/critical/modules/rds/security_groups.tf [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  ingress {
+    protocol  = "tcp"
+    from_port = 3306
+    to_port   = 3306
+
+    cidr_blocks = [
+      var.admin_cidr_ingress,
+    ]
+  }
```
Comment (alexwlchan) [inline infrastructure/critical/modules/rds/security_groups.tf]: Do we need to be able to access the database directly?

File: infrastructure/critical/modules/vhs/terraform.tf [hunk 4]
```diff
@@ -1,11 +1,11 @@
 data "terraform_remote_state" "shared_infra" {
   backend = "s3"
 
-  config {
+  config = {
     role_arn = "arn:aws:iam::760097843905:role/platform-developer"
```
Comment (alexwlchan) [inline infrastructure/critical/modules/vhs/terraform.tf]: While you're here, update this role as well? Tiny bit of permissions hygiene we’ve been gradually applying elsewhere:

```suggestion
    role_arn = "arn:aws:iam::760097843905:role/platform-read_only"
```

File: infrastructure/critical/terraform.tf [hunk 5]
```diff
@@ -14,20 +12,20 @@ terraform {
 data "terraform_remote_state" "shared_infra" {
   backend = "s3"
 
-  config {
+  config = {
     role_arn = "arn:aws:iam::760097843905:role/platform-developer"
```
Comment (alexwlchan) [inline infrastructure/critical/terraform.tf]: Ditto above.

File: infrastructure/critical/vhs_calm.tf [hunk 6]
```diff
@@ -1,6 +1,7 @@
-module "vhs_calm_sourcedata" {
-  source = "./vhs"
-  name   = "calm"
+module "vhs_calm" {
+  source = "./modules/vhs"
+  name   = "sourcedata-calm"
 
-  read_principals = ["${local.read_principles}"]
+  read_principals = local.read_principles
```
Comment (alexwlchan) [inline infrastructure/critical/vhs_calm.tf]: "principals ~> principles"

😭

File: infrastructure/critical/modules/rds/security_groups.tf (hunk 1 above)
Comment (alexwlchan) [reply infrastructure/critical/modules/rds/security_groups.tf]: PS Security descriptions are a massive faff to change after the fact, so let's get it right now!

Comment (jamesgorrie) [reply infrastructure/critical/modules/rds/security_groups.tf]: > Does it actually, or have you copy/pasted this without checking it?

Correcto

File: infrastructure/critical/modules/rds/security_groups.tf [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    security_groups = var.db_access_security_group
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
```
Comment (jamesgorrie) [inline infrastructure/critical/modules/rds/security_groups.tf]: @alexwlchan actually on this - I might not change anything, the reason being that will update the state, and thus the RDS instance, which, honestly, I don't want to do given the sensitive nature of the data.

Comment (jamesgorrie) [reply infrastructure/critical/modules/rds/security_groups.tf]: Take it back, it doesn't force and remaking of resources.

File: infrastructure/critical/modules/rds/security_groups.tf (hunk 3 above)
Comment (jamesgorrie) [reply infrastructure/critical/modules/rds/security_groups.tf]: Assuming for debugging or something. This I might not change given it's the ID minter

File: infrastructure/critical/modules/rds/security_groups.tf (hunk 1 above)
Comment (jamesgorrie) [reply infrastructure/critical/modules/rds/security_groups.tf]: Updated, does that make more sense?

**Pull request merged**